sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -390,6 +390,37 @@ void getUSMHostOrDevicePtr(PtrT usm_ptr, CUmemorytype *out_mem_type,
   }
 }
 
+bool getMaxRegistersJitOptionValue(const std::string &build_options,
+                                   unsigned int &value) {
+  using namespace std::string_view_literals;
+  const std::size_t optionPos = build_options.find_first_of("maxrregcount"sv);
+  if (optionPos == std::string::npos) {
+    return false;
+  }
+
+  bool valid = false;
+  const std::size_t delimPos = build_options.find('=', optionPos + 1u);
+  const std::size_t startPos = delimPos + 1;
+  if (delimPos != std::string::npos && startPos < build_options.length()) {
+    // validation function helper for value characters
+    static auto is_digit = [](char c) -> bool {
+      return std::isdigit(static_cast<unsigned char>(c)) != 0;
+    };
+
+    std::size_t pos = startPos;
+    while (build_options[pos] != '\n' && is_digit(build_options[pos])) {
+      pos++;
+    }
+    const std::string valueString =
+        build_options.substr(startPos, pos - startPos);
+    if (!valueString.empty()) {
```

Comment:
Nit; Likewise, if you just exit here you can avoid the need for `valid` entirely.